clients/src/test/java/org/apache/kafka/common/requests/EndTxnRequestTest.java

```diff
@@ -54,4 +57,61 @@ public void testConstructor() {
             assertEquals(throttleTimeMs, response.throttleTimeMs());
         }
     }
+
+    @Test
+    public void testEndTxnRequestWithTransactionsV2Enabled() {
```

Comment:
could we parameterize this test? seems like the main difference is th isTransactionV2Enabled (could parameterize on that) and then the version expected.